tools/check/check.go

```diff
@@ -129,33 +200,211 @@ func die(err error) {
 	dieWith(err, exitTool)
 }
 
+func parseConfig(path string) error {
+	if path == "" {
+		return nil
+	}
+	f, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("could not open %q: %v", path, err)
+	}
+	defer f.Close()
+
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return fmt.Errorf("could not read %q: %v", path, err)
+	}
+	if strings.HasSuffix(path, ".textproto") {
+		err = prototext.Unmarshal(contents, config)
+	} else {
+		err = proto.Unmarshal(contents, config)
+	}
+	if err != nil {
+		return fmt.Errorf("could not deserialize %q: %v", path, err)
+	}
+	// Populate fields that should not be nil
+	if config.RootOfTrust == nil {
+		config.RootOfTrust = &checkpb.RootOfTrust{}
+	}
+	if config.Policy == nil {
+		config.Policy = &checkpb.Policy{HeaderPolicy: &checkpb.HeaderPolicy{}, TdQuoteBodyPolicy: &checkpb.TDQuoteBodyPolicy{}}
+	}
+	return nil
+}
+
+func parsePaths(s string) ([]string, error) {
+	paths := strings.Split(s, ":")
+	if len(paths) == 1 && paths[0] == "" {
```

Comment:
Can we check the input string directly?
if (s == "") { return nil, nil }

Reply:
Done